### ISSUE TITLE
Calendar date UI broken when mixed with Bootstrap

### ISSUE THREAD

**Issue by ChaudhryAtif** [RenovoSolutions/ngx-datetimepicker#44]
Trying to use this picker along with Bootstrap 4.

![image](https://user-images.githubusercontent.com/7082580/31792289-1ffbddd2-b4e1-11e7-84b3-416abfe0bc84.png)

Not entirely sure what could be causing it. Perhaps some styling from BS4 is causing the UI to break.

**Comment by ChaudhryAtif** [comment]
Setting .calendar--day's width to 2.285em (instead of 1.285em) fixes UI issue with Bootstrap, but somewhat breaks without-bootstrap UI.

With BS4 and width 2.285em:
![image](https://user-images.githubusercontent.com/7082580/31797922-72d2c79a-b4f6-11e7-8d52-b77b3e6b9c6f.png)

W/o BS4 and width 2.285em:
![image](https://user-images.githubusercontent.com/7082580/31797890-48592e32-b4f6-11e7-969a-0ac6f7e7a9f8.png)

**Comment by ChaudhryAtif** [comment]
Adding/updating these classes made it work with and without BS4:

```
.calendar--days-select {
  display: flex;
  flex-wrap: wrap;
  flex-direction: row;
  justify-content: space-between;
}

.calendar--day {
  padding: .5em 0;
  text-align: center;
  width: 2.25em;
  font-size: 1em;
}
```

---

@TheOriginalJosh
Btw, I'm making some UI changes to suit my needs. If you want, I can do a PR after I'm done? I can probably do a PR with some specifics if you only want some of the UI changes.

What I have so far: (without bootstrap applied)
![image](https://user-images.githubusercontent.com/7082580/31799849-4a912770-b502-11e7-9bcf-30e7ad5f2b19.png)
- A little compact compared to original version
- Bold selected period (AM, PM), and Month arrows (<>)
- Minute and hour selection has white background like Dates

![image](https://user-images.githubusercontent.com/7082580/31799974-22975cc0-b503-11e7-8468-78ddc3d2242b.png)
- Switched month and year view to match that of hour/minute

**Comment by JoshDSommer** [comment]
@ChaudhryAtif  a PR would be fantastic, without bootstrap applied does it still have the same look and feel? @BrentWMiller would you be able to review this PR?

**Comment by ChaudhryAtif** [comment]
@TheOriginalJosh The screenshots above are without Bootstrap applied. 

Do you want a PR for just the BS4 fix, or do you want me to include all of my UI changes as well? It may change things up for some people who are using the one-value-per-row view for Month and Year.

---

Also, if I wanted to add a font-awesome (or any other) dependency on my personal fork, would that go into ngx-datetimepicker/package.json or ngx-datetimepicker/ngx-datetimepicker/package.json

**Comment by BrentWMiller** [comment]
I can fix this to work with BS4. Just need to find the conflicting classes

**Comment by ChaudhryAtif** [comment]
@BrentWMiller I posted what I changed to get it working with and without BS4 in earlier reply. You can take a look at that..

**Comment by ChaudhryAtif** [comment]
@TheOriginalJosh @BrentWMiller 
#45 
- Fix BS4 issue
- Add disable feature (idea from [this](https://github.com/JojiePalahang/ngx-datetimepicker/commit/8bf613f7dec9d38f734b9cb4279b5f63b5a2acec) fork's commit)
- Add type='button' to prevent form submitting (from [this](https://github.com/bngams/ngx-datetimepicker/commit/4fd88b83ce7915d6cd3aac7c30ccffbbe5b566d1) fork's commit)

**Comment by KorinaGraber** [comment]
Fixed in 1.2.0 via #45
